lib/prepareFunctionsUpload.js

```diff
@@ -106,29 +90,23 @@ var _packageSource = function(options, tmpdir) {
     });
 
     reader.on('end', function() {
+      archive.append(JSON.stringify(configValues), { name: '.runtimeconfig.json' });
       archive.finalize();
-      try {
-        fs.removeSync(tmpdir.name);
-      } catch (e) {
-        utils.logWarning('Unable to delete temporary directory ' + tmpdir.name +
-          '. You may want to manually delete it to save disk space.');
-        logger.debug(e);
-      }
     });
   });
 };
 
 module.exports = function(context, options) {
-  var tmpdir;
-  return _prepareSource(context, options).then(function(result) {
-    tmpdir = result;
-
-    return parseTriggers(getProjectId(options), tmpdir.name);
+  var configValues;
+  var sourceDir = options.config.path(options.config.get('functions.source'));
+  return _prepareSource(context).then(function(result) {
```

Comment:
It's kinda weird that a function called `prepareSource` resolves with config values. Can we refactor this just a bit now that "preparing source" isn't copying things to a tmp dir?